### PR TITLE
VyOS: Add [vyos_l3_interfaces] resource module

### PR DESCRIPTION
**Pull request by NilashishC**
Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>
Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/168

##### SUMMARY
- Adds new module - `vyos_l3_interfaces`
- Targets model - https://github.com/ansible-network/resource_module_models/pull/13

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
vyos_l3_interfaces.py

### CODE REVIEW

File: test/integration/targets/vyos_l3_interfaces/tests/cli/overridden.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    msg: "START vyos_l3_interfaces merged integration tests on connection={{ ansible_connection }}"
+
+- include_tasks: _remove_config.yaml
+
+- include_tasks: _populate.yaml
+
+- name: Overrides all device configuration with provided configuration
+  vyos_l3_interfaces: &overridden
+    config:
+      - name: eth0
```
Comment (pabelanger) [inline test/integration/targets/vyos_l3_interfaces/tests/cli/overridden.yaml]: I don't believe we want to use this interface for testing, as it is mgmt

File: test/integration/targets/vyos_l3_interfaces/vars/main.yaml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+  - name: "eth2"
+    ipv4:
+      - address: "192.0.2.10/24"
+      - address: "192.0.2.11/24"
+    ipv6:
+      - address: "2001:db8::10/32"
+      - address: "2001:db8::12/32"
+
+  - name: "eth0"
```
Comment (pabelanger) [inline test/integration/targets/vyos_l3_interfaces/vars/main.yaml:53]: we should avoid changing this

File: test/integration/targets/vyos_l3_interfaces/vars/main.yaml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  after:
+    - name: "eth2"
+      ipv6:
+        - address: "2001:db8::11/32"
+
+    - name: "eth1"
+      ipv4:
+        - address: "192.0.2.19/24"
+
+    - name: "eth0"
```
Comment (pabelanger) [inline test/integration/targets/vyos_l3_interfaces/vars/main.yaml:76]: same

File: test/integration/targets/vyos_l3_interfaces/vars/main.yaml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  commands:
+    - delete interfaces ethernet eth1 address '192.0.2.14/24'
+    - set interfaces ethernet eth1 address '192.0.2.15/24'
+    - delete interfaces ethernet eth2 address '192.0.2.10/24'
+    - delete interfaces ethernet eth2 address '192.0.2.11/24'
+    - delete interfaces ethernet eth2 address '2001:db8::10/32'
+    - delete interfaces ethernet eth2 address '2001:db8::12/32'
+
+  after:
+    - name: "eth0"
```
Comment (pabelanger) [inline test/integration/targets/vyos_l3_interfaces/vars/main.yaml:90]: same

File: test/integration/targets/vyos_l3_interfaces/tests/cli/overridden.yaml (hunk 1 above)
Comment (NilashishC) [reply test/integration/targets/vyos_l3_interfaces/tests/cli/overridden.yaml]: @pabelanger The reason for mentioning this interface in the task is to make sure that overridden doesn't cleanup eth0's IP configurations. This task, in fact, doesn't execute any command to modify eth0.

File: test/integration/targets/vyos_l3_interfaces/vars/main.yaml (hunk 2 above)
Comment (NilashishC) [reply test/integration/targets/vyos_l3_interfaces/vars/main.yaml:53]: This is the variable's file for assertion.

File: test/integration/targets/vyos_l3_interfaces/vars/main.yaml (hunk 3 above)
Comment (NilashishC) [reply test/integration/targets/vyos_l3_interfaces/vars/main.yaml:76]: This is the variable's file for assertion.

File: test/integration/targets/vyos_l3_interfaces/vars/main.yaml (hunk 4 above)
Comment (NilashishC) [reply test/integration/targets/vyos_l3_interfaces/vars/main.yaml:90]: This is the variable's file for assertion.

File: docs/docsite/rst/porting_guides/porting_guide_2.9.rst [hunk 5]
```diff
@@ -65,6 +65,8 @@ The following modules will be removed in Ansible 2.13. Please update update your
 * nxos_linkagg use :ref:`nxos_lag_interfaces <nxos_lag_interfaces_module>` instead.
 
 * vyos_interface use :ref:`vyos_interfaces <vyos_interfaces_module>` instead.
+ 
+* vyos_l3_interface use :ref:`vyos_l3_interfaces <vyos_l3_interfaces_module>` instead.
```
Comment (danielmellado) [inline docs/docsite/rst/porting_guides/porting_guide_2.9.rst:69]: removed or just deprecated for now?

File: lib/ansible/modules/network/vyos/_vyos_l3_interface.py [hunk 6]
```diff
@@ -33,6 +33,10 @@
 description:
   - This module provides declarative management of L3 interfaces
     on VyOS network devices.
+deprecated:
+  removed_in: '2.13'
```
Comment (danielmellado) [inline lib/ansible/modules/network/vyos/_vyos_l3_interface.py:37]: same question as in the previous commit.

**Review by danielmellado — APPROVED** [review]
LGTM. Thanks Nilashish! Just some comment there!

File: docs/docsite/rst/porting_guides/porting_guide_2.9.rst (hunk 5 above)
Comment (NilashishC) [reply docs/docsite/rst/porting_guides/porting_guide_2.9.rst:69]: @danielmellado deprecated for now.

File: lib/ansible/modules/network/vyos/_vyos_l3_interface.py (hunk 6 above)
Comment (NilashishC) [reply lib/ansible/modules/network/vyos/_vyos_l3_interface.py:37]: @danielmellado deprecated for now.

**Comment by NilashishC** [comment]
recheck

**Pull request merged**